src/Tizen.Multimedia.MediaPlayer/Player/Player.cs

```diff
@@ -345,6 +345,69 @@ public virtual Task PrepareAsync()
                 TaskScheduler.Default);
         }
 
+        /// <summary>
+        /// Prepares the cancellable media player for playback, asynchronously.
+        /// </summary>
+        /// <param name="cancellationToken">The cancellation token to cancel preparing.</param>
+        /// <seealso cref="CancellationToken"/>
+        /// <returns>A task that represents the asynchronous prepare operation.</returns>
+        /// <remarks>To prepare the player, the player must be in the <see cref="PlayerState.Idle"/> state,
+        /// and a source must be set.
+        /// The state must be <see cref="PlayerState.Preparing"/> to cancel preparing.
+        /// When preparing is cancelled, a state will be changed to <see cref="PlayerState.Idle"/> from <see cref="PlayerState.Preparing"/>.
+        /// <exception cref="ObjectDisposedException">The player has already been disposed of.</exception>
+        /// <exception cref="InvalidOperationException">
+        ///     Operation failed; internal error.
+        ///     -or-<br/>
+        ///     The player is not in the valid state.
+        ///     </exception>
+        /// <seealso cref="PrepareAsync()"/>
+        /// <seealso cref="Unprepare()"/>
+        /// <seealso cref="Prepared"/>
+        /// <since_tizen> 6 </since_tizen>
```

Comment:
I wonder why do not replace implementation of `public virtual Task PrepareAsync()` method with this method like below
``` c#
 public virtual Task PrepareAsync() {
     return PrepareAsync(null);
 }
```

Reply:
CancellationToken is `struct` type. So, if you call like `PrepareAsync(null)`, it's build break.
VS warn you with following message.
```
cannot convert from `<null>` to CancellationToken.
```

Reply:
Yes, you are right, it can't be null but CancellationToken provide a empty token to use as default
``` c#
 return PrepareAsync(CancellationToken.None);
```
https://docs.microsoft.com/ko-kr/dotnet/api/system.threading.cancellationtoken.none?view=netframework-4.8#System_Threading_CancellationToken_None

Reply:
Right, it's better to change it like your comment, if there's no reason why it remained separately.

Reply:
actually i was trying to combine both methods. but i couldn't be sure if it is totally the same or not, because of "return Task.Factory.StartNew" from what i've read, it is dividing thread.
doens't have any issues about it?

Reply:
We need to check the thread of _prepareCallback that called by PrepareAsync
if _prepareCallback was called on main loop thread. it cause behavior break.

As is below code is working
``` c#
 player.PrepareAsync().Wait();
```
But, 
if _prepareCallback was only execute on main loop thread, 
PrepareAsync(CancellationToken cancellationToken) is not working on same behavior
``` c#
 player.PrepareAsync(CancellationToken.None).Wait();
```

Reply:
while Prepare() in PrepareAsync() is executed on a new thread, _prepareCallback in PrepareAsync(CancellationToken) is called on the main thread.